tests/integration/utils2.go

```diff
@@ -1205,6 +1205,10 @@ func createDoc(
 		s.documents = append(s.documents, make([][]*client.Document, action.CollectionID-len(s.documents)+1)...)
 	}
 	s.documents[action.CollectionID] = append(s.documents[action.CollectionID], doc)
+
+	if action.ExpectedError == "" {
+		waitForUpdateEvents(s, action.NodeID, action.CollectionID)
```

Comment:
thought: This is a significant change in the tests - and one that looks like it removes our ability to test concurrent P2P stuff (something that we very much want/need to test)?

Reply:
The function name is a bit misleading, but `Update` events are published immediately after a document is saved and shouldn't really change how we test things concurrently. 

We could also merge `waitForUpdateEvents` into `waitForSync` since the events are buffered if you would prefer that.

Reply:
Ah I see what you mean - it is not waiting for the sync, just the event publishing to the current node's event queue?

Why are you calling it then?

Reply:
> Ah I see what you mean - it is not waiting for the sync, just the event publishing to the current node's event queue?

Yes

> Why are you calling it then?

Its a convenient way to track which doc heads are expected to be merged on remote nodes.

Reply:
> Its a convenient way to track which doc heads are expected to be merged on remote nodes.

Can you please document that really well?  Our integration tests are already super magic and far more complex than most, and that complexity is not really tested.  The P2P space is particularly complicated and prone to annoying, late-to-discover, flaky bugs and it is well worth over-explaining things to save future hassle.

Reply:
I added more documentation about how the document head are tracked. It should also be much easier to follow since its all moved to an `events.go` file. Let me know what you think.

Reply:
Changes look really good, thanks Keenan.  Has got me thinking about test Cid substitution too, in the future we might want to tweak `state.nodeP2P.DocHeads` to allow this.